jqm-all/jqm-model/src/main/java/com/enioka/jqm/jdbc/DbConn.java

```diff
@@ -595,4 +665,65 @@ public void logDatabaseInfo(Logger log)
     {
         return this.parent.getAdapter().poll(this, queue, nbSlots);
     }
+
+    static boolean testDbUnreachable(Exception e)
+    {
+        if ((e instanceof SQLTransientException) || (e.getCause() != null && e.getCause() instanceof SQLTransientException))
```

Comment:
This is not exactly what I was talking about. I had the idea to put the driver-specific part inside the adapter. Since most of these patterns only matter for a single DB type. Would this make sense to you?